docs/code/GraphRunner.md

```diff
@@ -0,0 +1,123 @@
+﻿# JSON Graph format
+
+The entry point graph in TLC is an array of _nodes_. Each node is an object with the following fields:
+
+- _name_: string. Required. Name of the entry point.
+- _inputs_: object. Optional. Specifies non-default inputs to the entry point. 
+Note that if the entry point has required inputs (which is very common), the _inputs_ field is requred.
+- _outputs_: object. Optional. Specifies the variables that will hold the node's outputs.
+
+## Input and output types
+The following types are supported in JSON graphs:
+
+- _string_. Represented as a JSON string, maps to a C# string.
+- _float_. Represented as a JSON float, maps to a C# float or double.
+- _bool_. Represented as a JSON bool, maps to a C# bool.
+- _enum_. Represented as a JSON string, maps to a C# enum. The allowed values are those of the C# enum (they are also listed in the manifest).
+- _int_. Currently not implemented. Represented as a JSON integer, maps to a C# int or long.
+- _array_ of the above. Represented as a JSON array, maps to a C# array.
+- _dictionary_. Currently not implemented. Represented as a JSON object, maps to a C# `Dictionary<string,T>`.
+- _component_. Currently not implemented. Represented as a JSON object with 2 fields: _name_:string and _settings_:object.
+
```

Comment:
Is this information current? I thought I saw some support for these. Certainly components are supported (not as `SubComponent` type specifically, but we can use dependency injection through the component factories).

Reply:
Ah, my edits to this file are not reflected. Fixing that.

Reply:
corrected the component part. Double-checking on the dictionaries and indexing in arrays. I don't think we do that yet. 

---
In reply to: [192866131](https://github.com/dotnet/machinelearning/pull/295#discussion_r192866131) [](ancestors = 192866131)